packages/graphql/src/index.ts

```diff
@@ -41,17 +40,13 @@ export type GraphQLClientQuery = {
  * ```
  */
 export const createGraphQLClient =
-  (
-    url: MaybeAccessor<string>,
-    headers?: RequestHeaders,
-    fetchFn?: typeof fetch
-  ): GraphQLClientQuery =>
+  (url: MaybeAccessor<string>, options?: RequestOptions): GraphQLClientQuery =>
   (query, variables: any = {}, initialValue) =>
     createResource(
       () => access(variables),
       (vars: any) => {
         const variables = typeof vars === "boolean" ? {} : vars;
-        return request(access(url), query, { headers, variables, fetcher: fetchFn });
+        return request(access(url), query, { variables, ...options });
```

Comment:
To prevent overwriting variables.
```suggestion
        return request(access(url), query, { ...options, variables });
```